### ISSUE TITLE
Canon Poor Exif Handling, Images Incorrectly Sorted

### ISSUE THREAD

**Issue by Civlac** [pulsejet/memories#985]
**Describe the bug**
- Sub-second time data is not accounted for when sorting Canon images, leading to some burst images being completely shuffled.
- The shutter speed is showing the "Shutter Speed Value", which can result in unfamiliar values. Should use "Exposure Time".

**To Reproduce**
- Image ordering
	- Worst case: File transfer images from local Windows 11 computer to Ubuntu Nextcloud instance via Samba. (Nextcloud configured to have External Local storage). Images can be several files out of place. (9480.jpg beside 9486.jpg)
	- Lesser extent: Upload images through Nextcloud Files web interface. This somehow orders most images better, but can swap neighbouring files' order. (9480.jpg after 9481.jpg)

**Platform:**
- OS: Windows
- Browser: Chrome
- Memories Version: 6.1.5
- Nextcloud Version: 27.1.5
- PHP Version: 8.2
- ExifTool Version Number: 12.70

**Additional context**
- For some reason, uploading the images through the online Nextcloud Files interface orders the images _better_ than moving them to Nextcloud Ext storage via Samba, despite the metadata appearing nearly identical, aside from "File Access Date/Time" and "File Inode Change Date/Time".
- All Canon models I've got my hands on have "Sub Sec Time" / "Sub Sec Original" / "Sub Sec Digitized", or "Create Date" / "Date/Time Original" / "Modify Date" that include subsecond data to the hundredths.
- [Additional detail for "Shutter Speed Value" vs "Exposure Time"](https://www.dpreview.com/forums/post/27645854)

**Feature Request**
- Within the Edit Metadata window, allow the user to modify the ms/subsecond value of timestamps to fix image sorting inconsistencies.

**Comment by Civlac** [comment]
Additionally, since not all cameras appear to have sub-second time data in the exif, has it been considered to make use of the filename for these instances?

**Comment by pulsejet** [comment]
Using the filename sounds like a good solution for now (though it wouldn't close this issue)

**Comment by Civlac** [comment]
Sounds good, and whether using Canon's Sub Sec metadata handling gets added in the future or not, other cameras, like an older Sony I looked at, doesn't have any Sub Second time data anywhere.
Using the filename as a fallback when multiple files have an equal date and time sounds like a good fix for all images, beyond the scope of Canon.

**Comment by pulsejet** [comment]
Basename is now used as a tie breaker for the date. Leave this issue open since it's not _really_ resolved.

**Comment by Civlac** [comment]
Thank you!
As for the second bug, do you wish to address it here or should I move it to a separate Issue?

**Comment by pulsejet** [comment]
Yup, it's just that the "ideal" solution would _also_ consider subsec metadata, then fall back to basename, then fall back to fileid (as earlier) etc.

**Comment by Civlac** [comment]
Sorry, I'm talking about the second bug I mentioned in the original post: The shutter speed is showing the "Shutter Speed Value", which can result in unfamiliar values. Should use "Exposure Time".
I supplied a link under the additional context if you wish to have some more info on it

**Comment by pulsejet** [comment]
Ah yes, that's a separate bug. I've zero knowledge about these things.

**Comment by Civlac** [comment]
No worries, in hindsight I should've made a seperate post about it 😁

Hopefully that one should be easy to swap over.
Maybe worth looking into what the "metadata" nextcloud app uses for shutter speed, just to have some confidence behind the change 👍

**Comment by LeonardFiedrowicz** [comment]
The metadata viewer still uses Shutter Speed Value instead of Exposure Time. 
Programs like Windows Photo Viewer or Photoshop or my camera itself always display Exposure time from my experience.
Some cameras have the same value in both fields so this might be a Canon specific issue?
